sympy/matrices/common.py

```diff
@@ -802,9 +804,8 @@ def entry(i, j):
 
     @classmethod
     def _eval_zeros(cls, rows, cols):
-        def entry(i, j):
-            return cls.zero
-        return cls._new(rows, cols, entry)
+        from sympy.matrices.dense import Matrix
+        return cls._new(Matrix._new(rows, cols, [S.Zero]*(rows*cols), copy = False))
```

Comment:
Why does this do `cls._new(Matrix._new`? Is `cls._new` not enough?

Probably it should also be `cls.zero`

Reply:
`return cls._new(rows, cols, [S.Zero]*(rows*cols), copy = False)` won't work as it will not have copy as argument.
For `./bin/test sympy/matrices/tests/test_commonmatrix.py`:
```
Traceback (most recent call last):
  File "c:\users\harshit yadav\sympy\sympy\matrices\tests\test_commonmatrix.py", line 861, in test_zeros
    assert list(SpecialOnlyMatrix.zeros(2, 2)) == [0, 0, 0, 0]
  File "c:\users\harshit yadav\sympy\sympy\matrices\common.py", line 1178, in zeros
    return klass._eval_zeros(rows, cols)
  File "c:\users\harshit yadav\sympy\sympy\matrices\common.py", line 806, in _eval_zeros
    return cls._new(rows, cols, [S.Zero]*(rows*cols), copy = False)
  File "c:\users\harshit yadav\sympy\sympy\matrices\common.py", line 3002, in _new
    return cls(*args, **kwargs)
TypeError: __init__() got an unexpected keyword argument 'copy'

==================== tests finished: 74 passed, 1 expected to fail, 1 exceptions, in 0.74 seconds =====================
DO *NOT* COMMIT!
```